packages/auth/src/signature/service.ts

```diff
@@ -124,6 +159,7 @@ async function verifySigFromBoundKey(
   return verifySigAndChallenge(deps, sig, sigInput, jwk, ctx)
 }
 
+// TODO: Replace with public httpsig library
 function sigInputToChallenge(sigInput: string, ctx: AppContext): string | null {
```

Comment:
I think it might be better to break this up into smaller functions like `getSigInputComponents`, `validateSigInputComponents`, and `generateChallengeFromSigInput`. Basically, `getSigInputComponents` is going to have to be called in both `validateSigInputComponents` (which would hold the code you're suggesting be moved) and `generateChallengeFromSigInput`.